### ISSUE TITLE
Sanity changes hash for SVG images

### ISSUE THREAD

**Issue by betalb** [sanity-io/sanity#3297]
**Describe the bug**

Sanity does some manipulation with SVG images, causing them to change contents. For already optimised SVG image those manipulations look like attributes reordering. The problem is that this is causing hash change, so it is possible to do dataset export and then import without changing cdn image URL

**To Reproduce**

1. Create a project with media plugin
2. Upload SVG image [original](https://user-images.githubusercontent.com/36754/169406979-ead4921f-5389-4a73-a7f5-0fab92e6484c.svg)
3. Download uploaded image [downloaded](https://user-images.githubusercontent.com/36754/169406974-bd5cdd58-0378-4a3b-b3be-724c10661eca.svg)
4. Compare hashes

**Expected behavior**

I would expect that hashes match

**Which versions of Sanity are you using?**

```
@sanity/cli             2.30.0 (up to date)
@sanity/base            2.30.0 (up to date)
@sanity/cli             2.30.0 (up to date)
@sanity/core            2.30.0 (up to date)
@sanity/default-layout  2.30.0 (up to date)
@sanity/default-login   2.30.0 (up to date)
@sanity/desk-tool       2.30.0 (up to date)
@sanity/vision          2.30.0 (up to date)
```

Similar issue was reported earlier https://github.com/sanity-io/sanity/issues/2321

**Comment by sourceful-david** [comment]
We are also seeing this issue too

**Comment by RitaDias** [comment]
👋 @sourceful-david @betalb 

Thank you for reporting this and sorry for taking so long to answer it.

I just had a quick look at this and from trying it with the most recent Sanity version it looks like this is not an issue anymore. At least, the svg code looked identical with the exception of moving some properties around (so the class might come after or before where it was on the original svg)

Closing the issue with this in mind. However, if you feel that the issue still exists, please re-open / recreate it. 
Once more, thanks :)

**Comment by betalb** [comment]
Good day @RitaDias

Actually what you described is the issue itself. Even though rendered svg is the same.

As far as I know, Sanity cdn uses hash to create permanent links and if during import attributes of svg are reordered, hashsum of the original svg will differ, so they will have different addresses. This is not happening to binary formats.

**Comment by RitaDias** [comment]
> Good day [@RitaDias](https://github.com/RitaDias)
> 
> Actually what you described is the issue itself. Even though rendered svg is the same.
> 
> As far as I know, Sanity cdn uses hash to create permanent links and if during import attributes of svg are reordered, hashsum of the original svg will differ, so they will have different addresses. This is not happening to binary formats.

Ahh you're right sorry, for some reason I didn't see that in the initial post but I see it now!
I'll re-open the issue 👍

**Comment by RitaDias** [comment]
Considering this is an upload and download difference, I presume this is something that would need to be tackeld in the content lake no? @judofyr thoughts?

**Comment by judofyr** [comment]
Yes. I believe this is a Content Lake issue. We’re sanitizing the SVG (from XSS and bad stuff) when uploading as opposed to when serving the file.

**Comment by RitaDias** [comment]
> Yes. I believe this is a Content Lake issue. We’re sanitizing the SVG (from XSS and bad stuff) when uploading as opposed to when serving the file.

@judofyr Since this is a content lake thing does it make sense to create a linear story (and to which team?) link this issue there? I'm not quite sure what to do with issues in these instances apart from leaving it open I suppose?

**Comment by kbrabrand** [comment]
This has been resolved now. We rolled out a fix for this earlier this week and starting from this week we keep the original in addition to the sanitized version. If you export you'll get the original file back, and that one will be the exact same file as you uploaded and match the hash.

Note that we still serve the optimized version, so if you download using the url you get when dereferencing an image asset reference in GROQ you'll get a sanitized asset back – this is the same as with other images.

This fix is not retroactive and SVGs uploaded before it was rolled out will still be the optimized version when exported.
